freegs/machine.py

```
@@ -442,6 +455,287 @@ def __ne__(self, other):
         return not self == other
 
 
+class Sensor:
+    def __init__(self, R, Z, name=None, weight=1, status=True,
+                 measurement=None):
+
+        self.R = R
+        self.Z = Z
+        self.status = status
+        self.name = name
+        self.weight = weight
+        self.measurement = measurement
+
+    def __repr__(self):
+        return "R={R}, Z={Z}".format(R=self.R, Z=self.Z)
+
+    def __eq__(self, other):
+        return np.allclose(self.R, other.R) and np.allclose(self.Z, other.Z)
+
+    def __ne__(self, other):
+        return not self == other
+
+
+class RogowskiSensor(Sensor):
+    """
+    Represents a Rogowski sensor.
+    Consists of an ordered list of (R,Z) points (defines polygon on which sensor lies)
+    Returns current inside the loop
+    """
+
+    def __init__(self, R, Z, current=0, name=None, weight=1, status=True,
+                 measurement=None):
+        Sensor.__init__(self, R, Z, name=name, weight=weight, status=status,
+                        measurement=measurement)
+
+    def to_numpy(self):
+        """
+        Helper method for writing output
+        """
+        return np.array((self.R, self.Z))
+
+    def get_measure(self, tokamak, equilibrium=None):
+        """
+        Method to update the current attribute of the sensor
+        with whatever current is contained within the sensor
+        if sensor is on (status == True)
+        """
+
+        if self.status:
```

self.status==False should probably raise an error if get_measure is called, or atleast 0, currently just leaves the old value in place.  maybe status should be static?  issue for all sensors